lesson_26/api/java/api_app/src/main/java/com/codedifferently/lesson26/web/MediaItemsController.java

```diff
@@ -22,11 +33,80 @@ public MediaItemsController(Library library) throws IOException {
+  @GetMapping()
+  public ResponseEntity<GetMediaItemsResponse> getItemById() {
     Set<MediaItem> items = library.search(SearchCriteria.builder().build());
+    if (items.isEmpty()) {
+      ResponseEntity.noContent();
+    }
     List<MediaItemResponse> responseItems = items.stream().map(MediaItemResponse::from).toList();
     var response = GetMediaItemsResponse.builder().items(responseItems).build();
-    return response;
+    return ResponseEntity.ok(response);
+  }
+
+  @GetMapping(value = "/{id}")
+  public ResponseEntity<MediaItemResponse> getItemById(@PathVariable("id") UUID id) {
+
+    System.out.println(id.toString());
+
+    Set<MediaItem> items = library.search(SearchCriteria.builder().id(id.toString()).build());
+    Optional<MediaItem> matchedItem =
+        items.stream().filter(item -> item.getId().equals(id)).findFirst();
+    System.out.println("items");
+    System.out.println(items);
+    System.out.println(matchedItem);
+    System.out.println("mathcedItems");
+    return matchedItem
+        .map(
+            item -> {
+              MediaItemResponse responseItem = MediaItemResponse.from(item);
+              return ResponseEntity.ok(responseItem);
+            })
+        .orElse(ResponseEntity.notFound().build());
+  }
+
+  @PostMapping()
+  public ResponseEntity<?> postItems(@RequestBody CreateMediaItemRequest requestBody) {
+    List<String> errorsList = new ArrayList<>();
+    MediaItemRequest request = requestBody.getItem();
+    if (requestBody.getItem() == null) {
+      errorsList.add("Cannot enter null");
+    } else {
+      if ((request.getTitle() == null || request.getTitle().isBlank())
+          && (request.getType() == null || request.getType().isBlank())) {
+        errorsList.add("title and type cannot be null or blank");
+      } else {
+        if (request.getTitle() == null || request.getTitle().isBlank()) {
+          errorsList.add("title cannot be null or blank");
+        } else {
+          if (request.getType() == null || request.getType().isBlank()) {
+            errorsList.add("type cannot be null or blank");
+          }
+        }
+      }
+    }
+    if (!errorsList.isEmpty()) {
+      return ResponseEntity.status(HttpStatus.BAD_REQUEST)
+          .body(
+              new Object() {
+                public final List<String> errors = errorsList;
+              });
+    }
+    MediaItem newMediaItem = MediaItemRequest.asMediaItem(request);
+    library.addMediaItem(newMediaItem, librarian);
+    return ResponseEntity.ok(requestBody);
+  }
+
+  @DeleteMapping(value = "/{id}")
+  public ResponseEntity<Void> deleteItem(@PathVariable("id") UUID id) {
+    Set<MediaItem> test1 = library.search(SearchCriteria.builder().build());
+    Set<MediaItem> item = library.search(SearchCriteria.builder().id(id.toString()).build());
+    if (item.isEmpty()) {
+      return ResponseEntity.notFound().build();
+    } else if (!test1.contains(item.iterator().next()) || item.iterator().next() == null) {
```

Comment:
Why are you doing this? Doesn't seem necessary at all. My guess is removing this code (and `test1`) above won't change your results at all.